### ISSUE TITLE
[REQUEST] Side By Side on PC only

### ISSUE THREAD

**Issue by lilgandhi1199** [jrbudda/minecrift#341]
How do I get into the menus and simply enable SBS mode without a VR headset?
I have a 3D projector plugged into my PC via hdmi that can decode 120hz 720p SBS into real 3D with my active IR glasses.
But it won't even let me boot into the game to switch to this mode.

What would the vivecraft options txt file look like to make this change without opening the game?

**Comment by ghost** [comment]
this software is not designed to drive a 3d output.  it is designed to render the buffer twice side by side which is how VR works, one render per eye.

**Comment by lilgandhi1199** [comment]
Perfect that's exactly what I described.
Let's make it happen

**Comment by ghost** [comment]
it is designed to drive the output to a vr headset which has its own proprietary firmware and driver, which is titled OpenVR.  to get it to work on an SBS projector, you need to rewrite the mod's code so that it no longer uses OpenVR and uses whatever library your projector requires instead.


tl;dr it's crashing because your projector is not supporting OpenVR.

**Comment by lilgandhi1199** [comment]
No it merely needs to leave the render on the computer.
When SBS is selected and a virtual emulator is ran is that not exactly what it does?
Or does it actually route it to the virtual input leaving the screen blank?

**Comment by ghost** [comment]
it routes one render (technically two) to what it is expecting to be a headset output loaded in by OpenVR, and then is rendering a mirror to your desktop.

again, you will need to rewrite this mod so that it does not use OpenVR whatsoever and goes desktop-screen-only instead.

**Comment by lilgandhi1199** [comment]
First off can you confirm SBS is a render option? Have you seen it yourself or tested it?
Secondly you are confirming that on the desktop it actually shows the side by side render itself? (while also duplicating it off to software)

Or is it merely showing the original 2D (left, right, or "center" eye)?

**Comment by ghost** [comment]
the SBS mode you are seeming to confuse for "Dual Eye Mirror mode" is just a mirror of what the "headset" is currently expected to be outputting, showing the left eye's buffer on the left, and the right eye's buffer on the right, stretched to fit the window.  because OpenVR has a bunch of overhead that checks for a headset so that it can interact with the firmware, it is seeing your projector as simply a projector and thus won't output to it.

for the third time, you will need to rewrite this mod to not use OpenVR in order to achieve what you are asking for.  OpenVR will never work with an SBS projector.

**Comment by lilgandhi1199** [comment]
Perfect.
There are plenty of virtual VR softwares I was just hoping to avoid all the BS overhead.
It seems I cannot. Still easy though lol
Peace

**Comment by ghost** [comment]
I would love to see how this fork turns out for you though as it is an interesting idea.  If you have any more questions about porting this mod to SBS projection, please visit the vivecraft discord.

**Comment by lilgandhi1199** [comment]
We'll see what kind of frame rate penalties I'll be getting for doing this duplication all the time...

No VR headset detected.
        Would you like to proceed anyways?
               Yes Please
:P

**Comment by ghost** [comment]
Vivecraft in Dual Eye Mirror mode renders the game 4 times.  so only asking for twice won't be a big issue.

**Comment by 46cv8** [comment]
@lilgandhi1199  You should perhaps try running vivecraft under https://github.com/oneup03/VRto3D it looks like it should support it but I haven't tried yet. Curious if it works though.

Just reporting back it works really well if you switch location mode to HMD and use vrto3d "yaw_enable" and "pitch_enable" for headset orientation emulation using the right joystick. The only issue is your pointer is pointing in some random place. fayer3 mentioned they might implement an aim option some day, at which point it will work perfectly and be usable for other than just flying around in 3d :) But still it seems the most full featured option for minecraft on 3d projectors and tv's that I could find.

**Comment by longlongCODE** [comment]
收到邮件了，但是如果邮件很重要还请你发一条短信通知我收信。号码18817566090    谢谢
